### DISCUSSION TITLE
Ziggy and Cypress Testing

### DISCUSSION THREAD

**Discussion by plweil** [tighten/ziggy#494] [Q&A]
Is anyone using Ziggy in a Laravel app on which they also do Cypress E2E testing? I run Cypress tests using several Valet URLs to simulate different kinds of users. With multiple domains, obviously I can't use the ziggy.js routes file from my regular environment. The only things I can think of so far are a) to use relative routes everywhere; or b) generate new ziggy.js route files in Cypress' `beforeEach()` hook. I'm not especially enamored with being forced to use relative routes. But generating special ziggy route files for each Cypress integration file feels messy and prevents me from cross-checking errors in my regular environment while testing. Has anyone tried anything else?

**Comment by bakerkretzmar** [comment]
Are you using the `@routes` Blade directive at all or just loading the routes from the generated file? I haven't ever used Cypress but would it work to generate the different route files you need in advance, one-per-domain instead of one-per-test?

Then in your app (or maybe in the tests?) you could load the file based on the actual URL?

**Reply by plweil** [reply]
I'm not using `@routes`; just the generated file. Multiple files in advance would be great, but then I have to change the path in my vue.js file(s). Different sections of the site use a different `app.js` file; I'm having a hard time wrapping my mind around how to make that import (`import { Ziggy } from '@/ziggy'`) global or dynamic (or both). For now, at least, I'm generating a new file in the `before()` hook for each group of tests. I thought making all routes relative might work, but for some reason either Cypress or something else can't let go of the `base_url`.

**Reply by plweil** [reply]
I take it back. Forcing relative routes is working; it's not necessary to regenerate the Ziggy routes file when running tests using different domains.
